test/integration/patients/worklist/schedule.js

```diff
@@ -1895,11 +1901,20 @@ context('schedule page', function() {
       .navigate('/worklist');
 
     cy
-      .go('back')
-      .then(() => {
-        const storage = JSON.parse(localStorage.getItem(`schedule_11111_11111-${ STATE_VERSION }`));
+      .go('back');
 
-        expect(storage.lastSelectedIndex).to.equal(null);
-      });
+    cy
+      .get('@scheduleList')
+      .find('.schedule-list__list-row')
+      .first()
+      .find('.schedule-list__day-list-row')
+      .eq(2)
+      .find('.js-select')
+      .click({ shiftKey: true });
+
+    cy
+      .get('@scheduleList')
+      .find('.schedule-list__day-list-row.is-selected')
+      .should('have.length', 2);
   });
 });
```

Comment:
In this test I removed all the checks for the `localstorage.lastSelectedIndex` we were doing and updated/added portions of it to ensure the `state.lastSelectedIndex` values were being set/cleared properly.

The test now does the following:

1. Shift+click multi-select from the top of the list to the bottom.
2. Shift+click multi-select the bottom of the list to the top.
3. Verify `state.lastSelectedIndex` is cleared when a user un-selects a list item.
4. Verify `state.lastSelectedIndex` is cleared when the `Clear` button is clicked.
5. Verify `state.lastSelectedIndex` is cleared when the `Find in list...` input is used to filter the list.
6. Verify `state.lastSelectedIndex` is cleared when a user navigates away from the page and comes back.